module/validation/approval_validator.go

```diff
@@ -0,0 +1,76 @@
+package validation
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/state/protocol"
+	"github.com/onflow/flow-go/storage"
+)
+
+type approvalValidator struct {
+	state    protocol.State
+	verifier module.Verifier
+}
+
+func NewApprovalValidator(state protocol.State, verifier module.Verifier) *approvalValidator {
+	return &approvalValidator{
+		state:    state,
+		verifier: verifier,
+	}
+}
+
+func (v *approvalValidator) Validate(approval *flow.ResultApproval) error {
+	// check if we already have the block the approval pertains to
+	head, err := v.state.AtBlockID(approval.Body.BlockID).Head()
+	if err != nil {
+		if !errors.Is(err, storage.ErrNotFound) {
+			return fmt.Errorf("failed to retrieve header for block %x: %w", approval.Body.BlockID, err)
+		}
+		return engine.NewOutdatedInputErrorf("no header for block: %v", approval.Body.BlockID)
+	} else {
```

Comment:
Lets remove the `else` here:
* as we always return for a satisfied if conditional, the else is not required.